content/blog/2022-07-26-schema-evolution-glue-msk/index.md

```diff
@@ -0,0 +1,861 @@
+
+Kafka clusters are often used as the central messaging infrastructure in complex microservice environments.
+However, the continuous and independent development of the individual microservices - the data producers and consumers - can make it hard to coordinate and evolve data schemas over time without introducing application failures due to incompatibilities.
+A common solution to this problem is to use a schema registry which provides for the validation of schema changes, preventing any unsafe changes and subsequent application failures.
+
+[AWS Glue Schema Registry](https://docs.aws.amazon.com/glue/latest/dg/schema-registry.html) can be used as such a schema registry, enabling you to validate and evolve streaming data using Apache Avro schemas.
+It can be easily integrated into Java applications for Apache Kafka with [AWS' official open-source serializers and deserializers](https://github.com/awslabs/aws-glue-schema-registry).
+
+<div id="integration-description"></div>
+The following chart shows the integration of producers and consumers with Amazon MSK and the AWS Glue Schema Registry:
+
+{{< figure src="schema-evolution-glue-msk-flow.svg" width="100%" alt="Workflow: Glue Schema Registry with MSK">}}
+
+1. Before sending a record, the producer validates that the schema it is using to serialize its records is valid. If the schema is not yet registered, the producer can be configured to automatically register a new schema version.
+   - When registering the new schema version, the schema registry validates if the schema is compatible.
+   - If the registry detects an incompatibility, the registration is rejected. This ensures that a producer fails early and cannot publish incompatible records in the first place.
+2. Once the schema is known to be valid, the producer serializes and compresses the record, and sends it to the Kafka cluster.
+3. The serialized and compressed record is read by the consumers.
+4. The consumer requests the schema from the schema registry (if it is not already cached), and uses the schema to decompress and deserialize the record.
+
+[AWS Glue Schema Registry is supported by LocalStack Pro as well](https://docs.localstack.cloud/aws/glue/), ultimately allowing you to test the evolution of your data streaming application completely on your local machine.
+
+# TL/DR
+- LocalStack Pro supports Amazon Managed Streaming for Apache Kafka (MSK) and AWS Glue Schema Registry.
+- This allows you develop and test your application's data schema evolution locally.
+- The code for this tutorial (including a script to execute it step-by-step) can be found in our [localstack-pro-samples repository on GitHub](https://github.com/localstack/localstack-pro-samples/tree/master/glue-msk-schema-registry).
+
+# Prerequisites
+For this tutorial you will need:
+- [LocalStack Pro](https://localstack.cloud/pricing/) to emulate Amazon MSK and AWS Glue Schema Registry locally
```

Comment:
Can we direct users to: app.localstack.cloud instead? A pricing page might just intimidate them.

Reply:
`app.localstack.cloud` directly shows a login barrier without any additional information. I personally prefer the pricing page which mentions the trial.
But that's not a strong opinion. ;)
Should I change the link?